File: openapi-generator/dart/lib/model/vehicle_safety_score_response.dart
part of openapi.api;

class VehicleSafetyScoreResponse {
  /* Crash event count */
  int crashCount = null;
  /* Harsh acceleration event count */
  int harshAccelCount = null;
  /* Harsh braking event count */
  int harshBrakingCount = null;
  
  List<SafetyReportHarshEvent> harshEvents = [];
  /* Harsh turning event count */
  int harshTurningCount = null;
  /* Safety Score */
  int safetyScore = null;
  /* Safety Score Rank */
  String safetyScoreRank = null;
  /* Amount of time driven over the speed limit in milliseconds */
  int timeOverSpeedLimitMs = null;
  /* Total distance driven in meters */
  int totalDistanceDrivenMeters = null;
  /* Total harsh event count */
  int totalHarshEventCount = null;
  /* Amount of time driven in milliseconds */
  int totalTimeDrivenMs = null;
  /* Vehicle ID */
  int vehicleId = null;
  VehicleSafetyScoreResponse();

  @override
  String toString() {
    return 'VehicleSafetyScoreResponse[crashCount=$crashCount, harshAccelCount=$harshAccelCount, harshBrakingCount=$harshBrakingCount, harshEvents=$harshEvents, harshTurningCount=$harshTurningCount, safetyScore=$safetyScore, safetyScoreRank=$safetyScoreRank, timeOverSpeedLimitMs=$timeOverSpeedLimitMs, totalDistanceDrivenMeters=$totalDistanceDrivenMeters, totalHarshEventCount=$totalHarshEventCount, totalTimeDrivenMs=$totalTimeDrivenMs, vehicleId=$vehicleId, ]';
  }

  VehicleSafetyScoreResponse.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['crashCount'] == null) {
      crashCount = null;
    } else {
          crashCount = json['crashCount'];
    }
    if (json['harshAccelCount'] == null) {
      harshAccelCount = null;
    } else {
          harshAccelCount = json['harshAccelCount'];
    }
    if (json['harshBrakingCount'] == null) {
      harshBrakingCount = null;
    } else {
          harshBrakingCount = json['harshBrakingCount'];
    }
    if (json['harshEvents'] == null) {
      harshEvents = null;
    } else {
      harshEvents = SafetyReportHarshEvent.listFromJson(json['harshEvents']);
    }
    if (json['harshTurningCount'] == null) {
      harshTurningCount = null;
    } else {
          harshTurningCount = json['harshTurningCount'];
    }
    if (json['safetyScore'] == null) {
      safetyScore = null;
    } else {
          safetyScore = json['safetyScore'];
    }
    if (json['safetyScoreRank'] == null) {
      safetyScoreRank = null;
    } else {
          safetyScoreRank = json['safetyScoreRank'];
    }
    if (json['timeOverSpeedLimitMs'] == null) {
      timeOverSpeedLimitMs = null;
    } else {
          timeOverSpeedLimitMs = json['timeOverSpeedLimitMs'];
    }
    if (json['totalDistanceDrivenMeters'] == null) {
      totalDistanceDrivenMeters = null;
    } else {
          totalDistanceDrivenMeters = json['totalDistanceDrivenMeters'];
    }
    if (json['totalHarshEventCount'] == null) {
      totalHarshEventCount = null;
    } else {
          totalHarshEventCount = json['totalHarshEventCount'];
    }
    if (json['totalTimeDrivenMs'] == null) {
      totalTimeDrivenMs = null;
    } else {
          totalTimeDrivenMs = json['totalTimeDrivenMs'];
    }
    if (json['vehicleId'] == null) {
      vehicleId = null;
    } else {
          vehicleId = json['vehicleId'];
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'crashCount': crashCount,
      'harshAccelCount': harshAccelCount,
      'harshBrakingCount': harshBrakingCount,
      'harshEvents': harshEvents,
      'harshTurningCount': harshTurningCount,
      'safetyScore': safetyScore,
      'safetyScoreRank': safetyScoreRank,
      'timeOverSpeedLimitMs': timeOverSpeedLimitMs,
      'totalDistanceDrivenMeters': totalDistanceDrivenMeters,
      'totalHarshEventCount': totalHarshEventCount,
      'totalTimeDrivenMs': totalTimeDrivenMs,
      'vehicleId': vehicleId
    };
  }

  static List<VehicleSafetyScoreResponse> listFromJson(List<dynamic> json) {
    return json == null ? new List<VehicleSafetyScoreResponse>() : json.map((value) => new VehicleSafetyScoreResponse.fromJson(value)).toList();
  }

  static Map<String, VehicleSafetyScoreResponse> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, VehicleSafetyScoreResponse>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new VehicleSafetyScoreResponse.fromJson(value));
    }
    return map;
  }
}

File: swagger-codegen/csharp-dotnet2/src/main/CsharpDotNet2/samsara/Model/VehicleSafetyScoreResponse.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace samsara.Model {

  /// <summary>
  /// Safety score details for a vehicle
  /// </summary>
  [DataContract]
  public class VehicleSafetyScoreResponse {
    /// <summary>
    /// Crash event count
    /// </summary>
    /// <value>Crash event count</value>
    [DataMember(Name="crashCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "crashCount")]
    public int? CrashCount { get; set; }

    /// <summary>
    /// Harsh acceleration event count
    /// </summary>
    /// <value>Harsh acceleration event count</value>
    [DataMember(Name="harshAccelCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "harshAccelCount")]
    public int? HarshAccelCount { get; set; }

    /// <summary>
    /// Harsh braking event count
    /// </summary>
    /// <value>Harsh braking event count</value>
    [DataMember(Name="harshBrakingCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "harshBrakingCount")]
    public int? HarshBrakingCount { get; set; }

    /// <summary>
    /// Gets or Sets HarshEvents
    /// </summary>
    [DataMember(Name="harshEvents", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "harshEvents")]
    public List<SafetyReportHarshEvent> HarshEvents { get; set; }

    /// <summary>
    /// Harsh turning event count
    /// </summary>
    /// <value>Harsh turning event count</value>
    [DataMember(Name="harshTurningCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "harshTurningCount")]
    public int? HarshTurningCount { get; set; }

    /// <summary>
    /// Safety Score
    /// </summary>
    /// <value>Safety Score</value>
    [DataMember(Name="safetyScore", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "safetyScore")]
    public int? SafetyScore { get; set; }

    /// <summary>
    /// Safety Score Rank
    /// </summary>
    /// <value>Safety Score Rank</value>
    [DataMember(Name="safetyScoreRank", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "safetyScoreRank")]
    public string SafetyScoreRank { get; set; }

    /// <summary>
    /// Amount of time driven over the speed limit in milliseconds
    /// </summary>
    /// <value>Amount of time driven over the speed limit in milliseconds</value>
    [DataMember(Name="timeOverSpeedLimitMs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "timeOverSpeedLimitMs")]
    public int? TimeOverSpeedLimitMs { get; set; }

    /// <summary>
    /// Total distance driven in meters
    /// </summary>
    /// <value>Total distance driven in meters</value>
    [DataMember(Name="totalDistanceDrivenMeters", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "totalDistanceDrivenMeters")]
    public int? TotalDistanceDrivenMeters { get; set; }

    /// <summary>
    /// Total harsh event count
    /// </summary>
    /// <value>Total harsh event count</value>
    [DataMember(Name="totalHarshEventCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "totalHarshEventCount")]
    public int? TotalHarshEventCount { get; set; }

    /// <summary>
    /// Amount of time driven in milliseconds
    /// </summary>
    /// <value>Amount of time driven in milliseconds</value>
    [DataMember(Name="totalTimeDrivenMs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "totalTimeDrivenMs")]
    public int? TotalTimeDrivenMs { get; set; }

    /// <summary>
    /// Vehicle ID
    /// </summary>
    /// <value>Vehicle ID</value>
    [DataMember(Name="vehicleId", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "vehicleId")]
    public int? VehicleId { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class VehicleSafetyScoreResponse {\n");
      sb.Append("  CrashCount: ").Append(CrashCount).Append("\n");
      sb.Append("  HarshAccelCount: ").Append(HarshAccelCount).Append("\n");
      sb.Append("  HarshBrakingCount: ").Append(HarshBrakingCount).Append("\n");
      sb.Append("  HarshEvents: ").Append(HarshEvents).Append("\n");
      sb.Append("  HarshTurningCount: ").Append(HarshTurningCount).Append("\n");
      sb.Append("  SafetyScore: ").Append(SafetyScore).Append("\n");
      sb.Append("  SafetyScoreRank: ").Append(SafetyScoreRank).Append("\n");
      sb.Append("  TimeOverSpeedLimitMs: ").Append(TimeOverSpeedLimitMs).Append("\n");
      sb.Append("  TotalDistanceDrivenMeters: ").Append(TotalDistanceDrivenMeters).Append("\n");
      sb.Append("  TotalHarshEventCount: ").Append(TotalHarshEventCount).Append("\n");
      sb.Append("  TotalTimeDrivenMs: ").Append(TotalTimeDrivenMs).Append("\n");
      sb.Append("  VehicleId: ").Append(VehicleId).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
